### PR TITLE
Fixing renames and dockerfile

### PR DESCRIPTION
**Pull request by jerryyin**
I noticed that docker auto builds are broken, so:
 - Fixed the job so it pulls from right repo
 - Noticed dockerfile is also broken because of referencing old repo name. Now it is stuck onto not able to build static lib, which it shouldn't in the first place.
 - Get rid of old repo name from comments

### CODE REVIEW

File: mlir/utils/jenkins/Dockerfile [hunk 1]
```diff
@@ -124,7 +124,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
 WORKDIR /MIOpenDepsWork
 RUN wget https://raw.githubusercontent.com/ROCmSoftwarePlatform/MIOpen/develop/requirements.txt \
     && wget https://raw.githubusercontent.com/ROCmSoftwarePlatform/MIOpen/develop/install_deps.cmake \
-    && sed -i -e '/llvm-project-mlir/d' requirements.txt \
+    && sed -i -e '/rocMLIR/d' requirements.txt \
     && cmake -P install_deps.cmake --prefix /usr/local
 RUN echo $(git ls-remote https://github.com/ROCmSoftwarePlatform/MIOpen HEAD) > rock-deps-commit-hash
```
Comment (zhanglx13) [inline mlir/utils/jenkins/Dockerfile]: This one is missing: `miopen-deps-commit-hash`

File: mlir/utils/jenkins/Dockerfile.rocm [hunk 2]
```diff
@@ -124,7 +124,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
 WORKDIR /MIOpenDepsWork
 RUN wget https://raw.githubusercontent.com/ROCmSoftwarePlatform/MIOpen/develop/requirements.txt \
     && wget https://raw.githubusercontent.com/ROCmSoftwarePlatform/MIOpen/develop/install_deps.cmake \
-    && sed -i -e '/llvm-project-mlir/d' requirements.txt \
+    && sed -i -e '/rocMLIR/d' requirements.txt \
     && cmake -P install_deps.cmake --prefix /usr/local
 RUN echo $(git ls-remote https://github.com/ROCmSoftwarePlatform/MIOpen HEAD) > rock-deps-commit-hash
```
Comment (zhanglx13) [inline mlir/utils/jenkins/Dockerfile.rocm]: same here

**Review by zhanglx13 — APPROVED** [review]
Some "not a must" missing renaming issues :)

File: mlir/utils/jenkins/Dockerfile (hunk 1 above)
Comment (jerryyin) [reply mlir/utils/jenkins/Dockerfile]: Yah, sounds good. Done in eafbbae

File: mlir/utils/jenkins/Dockerfile.rocm (hunk 2 above)
Comment (jerryyin) [reply mlir/utils/jenkins/Dockerfile.rocm]: I deleted this file instead. Somehow the dockerfile was duplicated twice in the renaming process.

**Review by sjw36 — APPROVED** [review]
Thanks @jerryyin !

**Review by krzysz00 — APPROVED** [review]
lgtm

**Pull request merged**